src/OpenCV.cc

```diff
@@ -224,8 +225,10 @@ NAN_METHOD(OpenCV::ReadImageAsync) {
         // async
         uint8_t *buf = (uint8_t *) Buffer::Data(Nan::To<v8::Object>(info[0]).ToLocalChecked());
         unsigned len = Buffer::Length(Nan::To<v8::Object>(info[0]).ToLocalChecked());
+        uint8_t *buf_new = (uint8_t *) malloc(len);
+        memcpy(buf_new, buf, len);
         Nan::Callback *callback = new Nan::Callback(cb.As<Function>());
-        Nan::AsyncQueueWorker(new AsyncImDecodeWorker(callback, buf, len, flags));
+        Nan::AsyncQueueWorker(new AsyncImDecodeWorker(callback, buf_new, len, flags));
```

Comment:
Instead of copying the buffer to mitigate premature garbage collection, it would be better to pass the object itself and make sure that it won't be garbage collected until `AsyncImDecodeWorker` has completed.

Reply:
Okay but i don't know how to do that.